tests/unit/dojson/test_dojson_hep.py

```diff
@@ -238,6 +235,344 @@ def test_authors(marcxml_to_json, json_to_marc):
+        '  <subfield code="m">gausyu@gmail.com</subfield>'
+        '  <subfield code="w">X.Gao.11</subfield>'
+        '  <subfield code="y">0</subfield>'
+        '</datafield>'
+    )  # record/1475380/export/xme
+
+    expected = [
+        {
+            'curated_relation': False,
+            'email': 'gausyu@gmail.com',
+            'full_name': 'Gao, Xu',
+            'ids': [
+                {
+                    'type': 'INSPIRE BAI',
+                    'value': 'X.Gao.11',
+                },
+            ],
+        },
+    ]
+    result = clean_record(hep.do(create_record(snippet)))
+
+    assert expected == result['authors']
+
+
+def test_authors_from_700__a_j_v_m_w_y():
+    snippet = (
+        '<datafield tag="700" ind1=" " ind2=" ">'
+        '  <subfield code="a">Liu, Ming</subfield>'
+        '  <subfield code="j">ORCID:0000-0002-3413-183X</subfield>'
+        '  <subfield code="v">School of Mathematics, South China University of Technology, Guangdong, Guangzhou, 510640, China</subfield>'
+        '  <subfield code="m">ming.l1984@gmail.com</subfield>'
+        '  <subfield code="w">M.Liu.16</subfield>'
+        '  <subfield code="y">0</subfield>'
+        '</datafield>'
+    )  # record/1475380/export/xme
+
+    expected = [
+        {
+            'curated_relation': False,
+            'email': 'ming.l1984@gmail.com',
+            'full_name': 'Liu, Ming',
+            'ids': [
+                {
+                    'type': 'ORCID',
+                    'value': '0000-0002-3413-183X',
+                },
+                {
+                    'type': 'INSPIRE BAI',
+                    'value': 'M.Liu.16',
+                },
+            ],
+        },
+    ]
+    result = clean_record(hep.do(create_record(snippet)))
+
+    assert expected == result['authors']
+
+
+@mock.patch('inspirehep.dojson.hep.fields.bd1xx.current_app.logger.error')
+def test_authors_from_700__double_a_u_w_x_y_z(error):
+    snippet = (
+        '<datafield tag="700" ind1=" " ind2=" ">'
+        '  <subfield code="a">Gumplinger, P.</subfield>'
+        '  <subfield code="a">Hadley, D.R.</subfield>'
+        '  <subfield code="u">Warwick U.</subfield>'
+        '  <subfield code="w">D.R.Hadley.2</subfield>'
+        '  <subfield code="x">1066999</subfield>'
+        '  <subfield code="y">0</subfield>'
+        '  <subfield code="z">903734</subfield>'
+        '</datafield>'
+    )  # record/1345256/export/xme
+
+    expected = [
+        {
+            'affiliations': [
+                {
+                    'record': {
+                        '$ref': 'http://localhost:5000/api/institutions/903734',
+                    },
+                    'value': 'Warwick U.',
+                },
+            ],
+            'curated_relation': False,
+            'full_name': 'Gumplinger, P.',  # XXX: wrong, but the best we can do.
```

Comment:
@kaplun also look at this test for an unwanted (but unavoidable) effect of `_get_full_name`.